scintillometry/io/psrfits/hdu.py

```diff
@@ -9,21 +9,43 @@
 from astropy.io import fits
 from astropy.utils import lazyproperty
 import numpy as np
+import operator
+from .psrfits_htm_parser import hdu_templates
 
 
 __all__ = ["HDU_map", "HDUWrapper", "PSRFITSPrimaryHDU",
            "SubintHDU", "PSRSubintHDU"]
 
 
 class HDUWrapper:
-    def __init__(self, hdu, verify=True):
+    def __init__(self, hdu=None, verify=True, hdu_type=None):
```

Comment:
Let's keep `verify` at the end (maybe make it keyword only by adding a `, *, verify=True`)